src/products/controllers/products/products.controller.ts

```diff
@@ -5,52 +5,29 @@ import {
     HttpStatus,
     Param,
     Post,
-    Patch,
     Res,
+    Put,
   } from '@nestjs/common';
-  import { ProductsService } from 'src/products/services/products/products.service';
-
-  
+import { ProductsService } from 'src/products/services/products/products.service';
+import { CreateProductDto } from 'src/products/dto/create-product.dto';
+import { UpdateProductDto } from 'src/products/dto/update-product.dto';
   @Controller('products')
   export class ProductsController {
     constructor(private readonly productService: ProductsService) {}
   
     @Post('add')
-    async createProduct(
-      @Res() response, 
-      @Body('productId') productId: string,
-      @Body('productType') productType: [],
-      @Body('gender') gender: [],
-      @Body('productShoeSize') productShoeSize: [],
-      @Body('productSize') productSize: [],
-      @Body('productSizePantsWaist') productSizePantsWaist: [],
-      @Body('productSizePantsInseam') productSizePantsInseam: [],
-      @Body('productDescriptionOptional') productDescriptionOptional: string,
-      @Body('productImage') productImage: string,
-
-      ) {
+    async createProduct(@Body() productDto: CreateProductDto) {
```

Comment:
Just for postman purposes, having a template of what fields are required would be helpful when testing the endpoints. 
ex: 
{
    "productId": "ABC123",
    "productType": [
        "Shoes"
    ],
...
}

Reply:
Other than that, it looks good! I tested the endpoints in postman and they're all working

Reply:
Thank you for your suggestion. 👍